### PR TITLE
(CAT-1317) - Add puppet module support

### PR DESCRIPTION
**Pull request by Ramesh7**
## Summary
Updating module with CAT standards.

## Additional Context
- [x] Rubocop 
- [x] Adding acceptance test
- [x] CI & Nightlies 
- [x] Cleanup
- [x] CODEOWNERS
- [x] PR Template

## Related Issues (if any)
N/A

## Checklist
- [x] 🟢 Spec tests.
- [x] 🟢 Acceptance tests.
- [x] Manually verified.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+# Offense count: 4
+# Configuration parameters: Severity, Include.
+# Include: **/*.gemfile, **/Gemfile, **/gems.rb
+Bundler/DuplicatedGem:
+  Exclude:
+    - 'Gemfile'
+
+# Offense count: 6
+# This cop supports safe autocorrection (--autocorrect).
```
Comment (jordanbreen28) [inline .rubocop_todo.yml:17]: should aim to correct all safe/unsafe rubocop autocorrections

**Review by GSPatton — APPROVED** [review]

File: .fixtures.yml [hunk 2]
```diff
@@ -1,3 +1,11 @@
 fixtures:
   repositories:
-    stdlib: https://github.com/puppetlabs/puppetlabs-stdlib
+    facts: 'https://github.com/puppetlabs/puppetlabs-facts.git'
+    puppet_agent:
+      repo: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
+      ref: v4.13.0
```
Comment (bastelfreak) [inline .fixtures.yml:6]: Why is that pinned?

File: .fixtures.yml [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+    stdlib:  https://github.com/puppetlabs/puppetlabs-stdlib.git
+    powershell: https://github.com/puppetlabs/puppetlabs-powershell.git
+    provision: 'https://github.com/puppetlabs/provision.git'
+  symlinks:
+    sslcertificate: "#{source_dir}"
```
Comment (bastelfreak) [inline .fixtures.yml:11]: the symlink isn't required. puppetlabs_spec_helper will create it for you automatically.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,5 @@
-**This module is deprecated. Do not use it anymore**
-
 # SSL Certificate module for Puppet
 
-[![Build Status](https://travis-ci.org/voxpupuli/puppet-sslcertificate.png?branch=master)](https://travis-ci.org/voxpupuli/puppet-sslcertificate)
-[![Puppet Forge](https://img.shields.io/puppetforge/v/puppet/sslcertificate.svg)](https://forge.puppetlabs.com/puppet/sslcertificate)
-[![Puppet Forge - downloads](https://img.shields.io/puppetforge/dt/puppet/sslcertificate.svg)](https://forge.puppetlabs.com/puppet/sslcertificate)
-[![Puppet Forge - endorsement](https://img.shields.io/puppetforge/e/puppet/sslcertificate.svg)](https://forge.puppetlabs.com/puppet/sslcertificate)
-[![Puppet Forge - scores](https://img.shields.io/puppetforge/f/puppet/sslcertificate.svg)](https://forge.puppetlabs.com/puppet/sslcertificate)
```
Comment (bastelfreak) [inline README.md:9]: is there a reason for droppng the forge links?

File: .fixtures.yml (hunk 2 above)
Comment (Ramesh7) [reply .fixtures.yml:6]: We are using same version across all the modules, so kept the convention here as well, as its pulling from github.

**Review by jordanbreen28 — APPROVED** [review]

**Pull request merged**